packages/core/host/networking/autohotspot/overlays/usr/libexec/autohotspot

```diff
@@ -12,29 +12,27 @@
 
 wifi_dev="wlan0" # device name to use; by default wlan0 is the on-board wifi chip on the RPi4
 
-# Determine the SSIDs of the wifi networks the RPi is configured to use:
-# FIXME: can we simplify these commands?
-wpa_ssid=$(grep -e '^[[:space:]]*ssid="' /etc/wpa_supplicant/wpa_supplicant.conf | \
-  awk -F'ssid=' '{ print $2 }' | sed 's/\r//g' | \
-  awk 'BEGIN{ORS=","} {print}' | sed 's/\"//g' | sed 's/,$//')
-IFS="," read -r -a ssids <<<"$wpa_ssid"
+hostname="$(sed 's~#.*$~~g' /etc/hostname | tr -d '[:space:]')"
+hostname_32="${hostname:0:32}"
+# list of ssids to connect to
+ssids=$(nmcli -t -f NAME,TYPE connection show | grep "802-11-wireless" | grep -v "$hostname_32" | cut -d':' -f1 | sort | uniq)
 
 start_isolated_wlan() {
   echo "Starting isolated wifi network..."
-  ip link set dev "$wifi_dev" down
-  ip address add 192.168.4.1/24 broadcast + dev "$wifi_dev"
-  ip link set dev "$wifi_dev" up
-  dhcpcd --release "$wifi_dev"
-  systemctl start hostapd
+  if nmcli connection show --active | grep -q "$hostname_32"; then
+    echo "Hotspot '$hostname_32' is already up"
+    return
+  fi
+  systemctl stop dnsmasq
+  sleep 5
```

Comment:
If you have to wait after running `systemctl stop dnsmasq`, can you replace `sleep 5` with a `while` loop to poll the status of dnsmasq and sleep for a shorter time (e.g. 1 second) if dnsmasq has not stopped yet?